portal-impl/src/com/liferay/portal/service/impl/PermissionPropagationEntryLocalServiceImpl.java

```diff
@@ -14,11 +14,47 @@
 
 package com.liferay.portal.service.impl;
 
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.PermissionPropagationEntry;
+import com.liferay.portal.kernel.util.Validator;
 import com.liferay.portal.service.base.PermissionPropagationEntryLocalServiceBaseImpl;
 
 /**
  * @author Brian Wing Shun Chan
+ * @author Quan Huynh
  */
 public class PermissionPropagationEntryLocalServiceImpl
 	extends PermissionPropagationEntryLocalServiceBaseImpl {
+
+	public void addPermissionPropagationEntry(
+			long groupId, long companyId, long classNameId, long classPK,
+			boolean propagation)
+		throws PortalException {
+
+		validate(classNameId, classPK);
+
+		long permissionPropagationEntryId = counterLocalService.increment();
+
+		PermissionPropagationEntry permissionPropagationEntry =
+			permissionPropagationEntryPersistence.create(
+				permissionPropagationEntryId);
+
+		permissionPropagationEntry.setGroupId(groupId);
+		permissionPropagationEntry.setCompanyId(companyId);
+		permissionPropagationEntry.setClassNameId(classNameId);
+		permissionPropagationEntry.setClassPK(classPK);
+		permissionPropagationEntry.setPropagation(propagation);
+
+		permissionPropagationEntryPersistence.update(
+			permissionPropagationEntry);
+	}
+
+	protected void validate(long classNameId, long classPK)
+		throws PortalException {
+
+		if (Validator.isNull(classNameId) || (classPK <= 0)) {
+			throw new PortalException();
+		}
+	}
```

Comment:
@ces-quanhuynh I  don't think we need to validate the classNameId and classPK.
Don't use Validator.isNull for long. it's never null.

Reply:
Thanks @tototrinh
I've updated the code. Please review again!